### PR TITLE
Create a KittySay Command

### PR DESCRIPTION
**Pull request by gserena01**
related to #13

### CODE REVIEW

**Comment by gserena01** [comment]
Strange, but line 14 of src/cmd_kittysay.rs will only run with line 15 present... I'm wondering what might be happening here?

**Comment by gserena01** [comment]
Also, @hanbollar if there was an image of KittyCAD without her shadow I think the image would read better as ASCII art.

**Comment by hanbollar** [comment]
@gserena01 - here's two options

just_the_kitty_small
just_the_kitty

**Comment by iterion** [comment]
> Strange, but line 14 of src/cmd_kittysay.rs will only run with line 15 present... I'm wondering what might be happening here?

Sorry, I totally missed this. I assume you're seeing an issue like this:
```
error[E0308]: mismatched types
  --> src/cmd_kittysay.rs:13:14
   |
13 |     async fn run(&self, ctx: &mut crate::context::Context) -> Result<()> {
   |              ^^^
   |              |
   |              expected enum `Result`, found `()`
   |              implicitly returns `()` as its body has no tail or `return` expression
   |
   = note:   expected enum `Result<(), anyhow::Error>`
           found unit type `()`
```

Basically, the return type of the function is the Result type. In particular, we use a convenient wrapper around the std called [anyhow](https://docs.rs/anyhow/latest/anyhow/).

If you only have the line:
```
        writeln!(ctx.io.out, "KITTYSAYSTUFF!");
```
Then the return type of the function is (). Rust has both implicit and explicit returns. If you end a function without a semicolon then the result of that statement is returned. 

To get this compiling without warnings we'd have to change it to:
```
        writeln!(ctx.io.out, "KITTYSAYSTUFF!")?;
        Ok(())
```

The `Ok(())` at the end matches the return type. You'll note the line above added a `?` which is a shortcut for an early return plus converting the error type. Let me know if you're still stuck.

**Comment by hanbollar** [comment]
@gserena01 - correct me if im wrong, but if `KITTYSAYSTUFF` is just a placeholder, make sure to swap it out with what the actual txt ascii is

overall lgtm, gonna leave it up to adam for final review on this (so no need to request me for review for this pr 👍 ) - removing the request

File: src/cmd_say.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+#[async_trait::async_trait]
+impl crate::cmd::Command for CmdSay {
+    async fn run(&self, ctx: &mut crate::context::Context) -> Result<()> {
+        let kitty_string;
+        if self.input.len() > 0 {
+            let text = self.input.join(" ");
+            let mut border = String::from("--");
+            let bar = String::from("|");
+            let print_text = format!("{}{}{}", bar, text, bar);
```
[the remark this answers is not in the document]
Comment (iterion) [reply src/cmd_say.rs:23]: You could remove the line above as well by including the `|` in the format.

File: src/cmd_say.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    /// What kitty says
+    #[clap(name = "input", required = false, multiple_values = true)]
+    pub input: Vec<String>,
+}
+
+#[async_trait::async_trait]
+impl crate::cmd::Command for CmdSay {
+    async fn run(&self, ctx: &mut crate::context::Context) -> Result<()> {
+        let kitty_string;
+        if self.input.len() > 0 {
```
Comment (iterion) [inline src/cmd_say.rs:19]: Follow the clippy lint.

File: src/cmd_say.rs [hunk 3]
```diff
@@ -0,0 +1,100 @@
+use anyhow::Result;
+use clap::Parser;
+
+/// Prints your text in a text bubble with KittyCAD as ASCII art
+///
+///     $ kittycad say
```
Comment (iterion) [inline src/cmd_say.rs:6]: We might want to add more examples here now that you can also give arguments.

File: src/cmd_say.rs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                :*#%%%%%%%%%%+       -%######*#+.         
+                =#########%%%+     =#%%%%%%%%%##-         
+                -++***#####=.      *############:         
+                                   -==++***##+:           
+";
+            writeln!(ctx.io.out, "{}", kitty_string).ok();
+        }
+        Ok(())
+    }
+}
```
Comment (iterion) [inline src/cmd_say.rs:100]: We should add at least one test. Check https://github.com/KittyCAD/cli/blob/main/src/cmd_file.rs#L431 and other files for some example test setup.

**Review by iterion — APPROVED** [review]

File: src/cmd_say.rs [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    #[clap(name = "input", required = false, multiple_values = true)]
+    pub input: Vec<String>,
+}
+
+#[async_trait::async_trait]
+impl crate::cmd::Command for CmdSay {
+    async fn run(&self, ctx: &mut crate::context::Context) -> Result<()> {
+        let kitty_speaking = self.input.len() > 0;
+        let kitty_string = format_kitty(kitty_speaking);
+        if self.input.len() > 0 {
```
Comment (iterion) [inline src/cmd_say.rs:22]: We can reuse `kitty_speaking` here. And above we should use `!self.input.is_empty()` like clippy suggests.

**Review by iterion — CHANGES_REQUESTED** [review]
Thanks for adding the tests! This looks good aside from just needing to run cargo fmt and following the clippy lints.

**Review by iterion — APPROVED** [review]
Assuming tests all pass this looks good to me!

**Comment by gserena01** [comment]
fixes #13

**Pull request merged**